### DISCUSSION TITLE
Another one cool update, problems with highlighting

### DISCUSSION THREAD

**Discussion by Mazan-ka** [dbeaver/dbeaver#22531] [Q&A]
I have updated DBeavear 23.3.3 today, and notice some troubles with strings highlighting. Attaching the scrincast

https://github.com/dbeaver/dbeaver/assets/80702083/f2be7cff-ba47-4ce7-ab45-2eb62abd4b8f

**Comment by E1izabeth** [comment]
Hi, @Mazan-ka. I suspect that you put excessive ' somewhere at the top part of the query which is hidden on the video.

**Reply by E1izabeth** [reply]
When you add new line, highlighting changes because by default new line is a statement delimiter. So, you turn your query to the two invalid queries.

**Reply by Mazan-ka** [reply]
This query is body of my function. 
In the previous version of DBeaver when I wanted to edit the value of a string variable, this line remained highlighted in green, regardless of what I changed. The selection only changed when I opened a new <'> mark or erased an old one, this is not the case now.

https://github.com/dbeaver/dbeaver/assets/80702083/bd5aa841-2b02-4dfb-88f9-8a86c40d58f4

**Reply by E1izabeth** [reply]
Get it, thank you for explanation.

**Reply by E1izabeth** [reply]
Unfortunately, I can't reproduce it. The green highlighting returns back almost immediately after editing in my case.
https://github.com/dbeaver/dbeaver/assets/28875055/e2e7ab69-2db2-49f7-8fc8-42d478201c70

Could you try switching these settings off and check if it helps?
![image](https://github.com/dbeaver/dbeaver/assets/28875055/d898b0e5-bdb8-4296-b937-3dde79cd8b22)

**Reply by E1izabeth** [reply]
Or do you have them switched off already?

**Reply by E1izabeth** [reply]
Oh, never mind, I reproduced the issue with disappearing of a green highlighting with the mentioned settings switched off.

**Reply by Mazan-ka** [reply]
Yes, I switched them off already because of this:
https://github.com/orgs/dbeaver/discussions/22262

**Reply by E1izabeth** [reply]
BTW, we fixed the issue with query validation. Could you try if this new highlighting works on the new version (DBeaver 23.3.3) for you?

**Reply by Mazan-ka** [reply]
Oh, I got the same situation as you in your screencast above, with this checkbox

**Reply by E1izabeth** [reply]
Hmm.. OK, thank you

**Comment by E1izabeth** [comment]
I created an issue #22542
Thank you for reporting
